assembly/runtime/promise.ts

```diff
@@ -0,0 +1,138 @@
+import { util } from "./util";
+import { env } from "./env";
+import { u128 } from ".";
+
+function encodeArgs<T>(t: T) : Uint8Array {
+  if (t instanceof Uint8Array) {
+    return t;
+  }
+  return encode<T>(t);
+}
+
+// TODO: not sure if these types make any sense at all
+type Balance = u128;
+type Gas = u64;
+type PublicKey = Uint8Array;
+type AccountId = Uint8Array;
+
+/**
+ * Batch actions from within an AssemblyScript contract
+ *
+ * (1) transfer money
+ *
+ * // assume context.sender = "bob.testnet"
+ * new ContractPromiseBatch("alice").transfer(100); // send 100 N to "alice"
+ *
+ * (2) deploy a contract
+ *
+ * // assume contract is loaded, either included and encoded as base64 in the current
+ * // contract itself or via input to a function call (requires some other changes in near-sdk-as)
+ * const code = Uint8Array[1,2,3]; // regardless, you end up with the contract code somehow
+ *
+ * // assume context.sender = "bob.testnet"
+ *
+ * ContractPromiseBatch
+ *     .create("app-v1.bob.testnet")
+ *     .create_account()
+ *     .transfer(u128.from(100))
+ *     .add_full_access_key(new Uint8Array(0)) // TODO: need signer public key as byte array
+ *     .deploy_contract(code)
+ *
+ * (3) make cross contract calls // TODO: this example doesn't make sense
```

Comment:
low confidence on the story behind this test, can you help improve?

Reply:
This is a silly example because it all could be done with normal promises.  A better test would be to create an account and then send it money.

Reply:
✅